tensorflow/lite/kernels/exp.cc

```diff
@@ -65,6 +101,34 @@ TfLiteStatus Eval(TfLiteContext* context, TfLiteNode* node) {
       case kTfLiteFloat32:
         TF_LITE_EXP(reference_ops, float);
         break;
+      case kTfLiteInt8: {
+        OpData* data = reinterpret_cast<OpData*>(node->user_data);
+        const TfLiteTensor* input = GetInput(context, node, 0);
+        TfLiteTensor* output = GetOutput(context, node, 0);
+        ExpParams params;
+        params.input_zero_point = input->params.zero_point;
+        params.input_range_radius = data->input_range_radius;
+        params.input_multiplier = data->input_multiplier;
+        params.input_left_shift = data->input_left_shift;
+        params.output_zero_point = output->params.zero_point;
+        reference_integer_ops::QIntExp(
+            params, GetTensorShape(input), GetTensorData<int8_t>(input),
+            GetTensorShape(output), GetTensorData<int8_t>(output));
+      } break;
+      case kTfLiteUInt8: {
+        OpData* data = reinterpret_cast<OpData*>(node->user_data);
+        const TfLiteTensor* input = GetInput(context, node, 0);
+        TfLiteTensor* output = GetOutput(context, node, 0);
+        ExpParams params;
+        params.input_zero_point = input->params.zero_point;
+        params.input_range_radius = data->input_range_radius;
+        params.input_multiplier = data->input_multiplier;
+        params.input_left_shift = data->input_left_shift;
+        params.output_zero_point = output->params.zero_point;
+        optimized_integer_ops::QUIntExp(
+            params, GetTensorShape(input), GetTensorData<uint8_t>(input),
+            GetTensorShape(output), GetTensorData<uint8_t>(output));
+      } break;
```

Comment:
The duplicated code is unfortunate. I would create a small helper in this file:


```
ExprParams GetExprParams(const OpData *data, const TFLiteTensor *input, const TFLiteTensor *output) {
  ExpParams params;
  params.input_zero_point = input->params.zero_point;
  params.input_range_radius = data->input_range_radius;
  params.input_multiplier = data->input_multiplier;
  params.input_left_shift = data->input_left_shift;
  params.output_zero_point = output->params.zero_point;
  return params;
}
```
then this code would become:

```

case kTfLiteInt8: {
  reference_integer_ops::QIntExp(
            GetExprParams(data, input, output), GetTensorShape(input), GetTensorData<int8_t>(input),
            GetTensorShape(output), GetTensorData<int8_t>(output));
} break;
case kTfLiteUInt8: {
        optimized_integer_ops::QUIntExp(
            GetExprParams(data, input, output), GetTensorShape(input), GetTensorData<uint8_t>(input),
            GetTensorShape(output), GetTensorData<uint8_t>(output));
} break;
```

Reply:
(but both QIntExp and QUintExp would be in reference_integer_ops namespace, as discussed above).

Reply:
@talumbau , i have accepted the comments and updated the code as per your suggestion.

Regards
Amit